internal/caravan/tradeagreement.lua

```diff
@@ -26,6 +37,59 @@ local function diplomacy_toggle_cat()
     end
 end
 
+local function OrderGemWithMinValue(gemPrices, minval)
+    local priority_idx = GetCurrentTabId()
+    if not IsTabGem(priority_idx) then
+        return
+    end
+
+    local priority = diplomacy.environment.meeting.sell_requests.priority[priority_idx]
+    for i in ipairs(priority) do
+        if gemPrices[i] >= minval then
+            priority[i] = 4
+        else
+            priority[i] = 0
+        end
+    end
+end
+
+local function GetCivGemPrices()
+    local civ_id = diplomacy.actor.civ_id
+    local civ = df.historical_entity.find(civ_id)
+    local gems = civ.resources.gems
+    local gemPrices = {}
+    local gemValuesUnique = {}
+    local filter = {}
+    for civid, matid in pairs(gems) do
+        local gem = dfhack.matinfo.decode(0, matid)
+        gemPrices[civid] = gem.material.material_value
+        local gemVal = gem.material.material_value
+        if not filter[gemVal] then
+            filter[gemVal] = true
+            table.insert(gemValuesUnique, gemVal)
+        end
+    end
+
+    table.sort(gemValuesUnique)
+    return gemPrices, gemValuesUnique
+end
+
+local function MinimumGemValue()
+    local gemPrices, gemValuesUnique = GetCivGemPrices()
+    local list = {}
+    for index, value in ipairs(gemValuesUnique) do
+        list[index] = tostring(value)
```

Comment:
just displaying the number is a little spartan. could we add some more information here? for example, it would be useful to know how many items are at each value tier.